src/source/internal.rs

```diff
@@ -106,312 +108,321 @@ macro_rules! atom_non_zero_telem {
-                    self.chans
-                );
-                atom_non_zero_telem!(
-                    "cernan.sinks.prometheus.write.binary",
-                    sink::prometheus::PROMETHEUS_WRITE_BINARY,
-                    self.tags,
-                    self.chans
-                );
-                atom_non_zero_telem!(
-                    "cernan.sinks.prometheus.write.text",
-                    sink::prometheus::PROMETHEUS_WRITE_TEXT,
-                    self.tags,
-                    self.chans
-                );
-                atom_non_zero_telem!(
-                    "cernan.sinks.prometheus.report_error",
-                    sink::prometheus::PROMETHEUS_REPORT_ERROR,
-                    self.tags,
-                    self.chans
-                );
-                // sink::influxdb
-                atom_non_zero_telem!(
-                    "cernan.sinks.influxdb.delivery_attempts",
-                    sink::influxdb::INFLUX_DELIVERY_ATTEMPTS,
-                    self.tags,
-                    self.chans
-                );
-                atom_non_zero_telem!(
-                    "cernan.sinks.influxdb.success",
-                    sink::influxdb::INFLUX_SUCCESS,
-                    self.tags,
-                    self.chans
-                );
-                atom_non_zero_telem!(
-                    "cernan.sinks.influxdb.failure.client_error",
-                    sink::influxdb::INFLUX_FAILURE_CLIENT,
-                    self.tags,
-                    self.chans
-                );
-                atom_non_zero_telem!(
-                    "cernan.sinks.influxdb.failure.server_error",
-                    sink::influxdb::INFLUX_FAILURE_SERVER,
-                    self.tags,
-                    self.chans
-                );
-                // filter::delay_filter
-                atom_non_zero_telem!(
-                    "cernan.filters.delay.telemetry.accept",
-                    filter::delay_filter::DELAY_TELEM_ACCEPT,
-                    self.tags,
-                    self.chans
-                );
-                atom_non_zero_telem!(
-                    "cernan.filters.delay.telemetry.reject",
-                    filter::delay_filter::DELAY_TELEM_REJECT,
-                    self.tags,
-                    self.chans
-                );
-                atom_non_zero_telem!(
-                    "cernan.filters.delay.log.reject",
-                    filter::delay_filter::DELAY_LOG_REJECT,
-                    self.tags,
-                    self.chans
-                );
-                atom_non_zero_telem!(
-                    "cernan.filters.delay.log.accept",
-                    filter::delay_filter::DELAY_LOG_ACCEPT,
-                    self.tags,
-                    self.chans
-                );
-                while let Some(mut telem) = Q.pop() {
+            let mut events = mio::Events::with_capacity(1024);
+            match poll.poll(& mut events, Some(slp)) {
```

Comment:
`poll.poll` will block, will it not? If it does, that's a problem for the steady beat of telemetry. 

If we're not catching a shutdown here should we be polling?

Reply:
It does block, with the same slp value. 

Same behavior as https://github.com/postmates/cernan/pull/339/files#diff-fa755baccca5ebdb28f49f30a86be0f7L112.

My expectation is that one day we will be catching system event here, hence it was plumbed in.